### PR TITLE
Derek/20230814-image-to-prompt-capability

### PR DESCRIPTION
**Pull request by 1orZero**
- [DONE] Add Supabase bucket / policy in Supabase (image-to-prompt) 
- Add Image to Prompt capability

Please let me know if there is any issue!
Screenshot 2023-08-16 at 9 26 10 PM

### CODE REVIEW

**Comment by williamtriinh** [comment]
Screenshot 2023-08-16 at 10 20 59

The layout doesn't seem to be accounted for for large images. The image in the screenshot is 1024x1024px.

**Comment by williamtriinh** [comment]
I'm also getting an error when I try to upload an image. I've already created the `image-to-prompt` bucket. 
Screenshot 2023-08-16 at 10 29 08

**Comment by 1orZero** [comment]
> I'm also getting an error when I try to upload an image. I've already created the `image-to-prompt` bucket. 
> 
> Screenshot 2023-08-16 at 10 29 08
> 
> 

To be able to test in your local environment, you can use Ngrok to expose your Supabase bucket. This will make the image URL accessible for the NextLeg API.

I will continue testing on the staging server tomorrow.

**Comment by 1orZero** [comment]
I have updated the README.md file for setting up "image-to-prompt". Please let me know if you encounter any issues.
@williamtriinh

**Comment by thejackwu** [comment]
@1orZero Works great! But can we only display the Image to prompt option when it's in AI image mode?

**Comment by 1orZero** [comment]
> @1orZero Works great! But can we only display the Image to prompt option when it's in AI image mode?

Done!

File: README.md [hunk 1]
```diff
@@ -71,4 +71,12 @@ Start chatting and enjoy!
...
\ No newline at end of file
+3. Update the `WEB_CONTENT_FUNCTION_ENDPOINT` after completing the initial deployment.
+
+# Set up image-to-prompt feature bucket for Local development 
+
+1. Create a supabase bucket `image-to-prompt` and make it public in the Local supabase server.
+2. Go to file `pages/api/image-to-prompt.ts` and uncomment the function `replaceLocalhost`
+3. Update the variable `ngrokHost ` and `localSupabaseHost` 
+4. Uncomment the line where url is passed to `url: replaceLocalhost(url)`, in function `nextLegDescribe`
+5. Now that your supabase bucket is accessible from the internet, so that the next leg api can access it.
```
Comment (thejackwu) [inline README.md:82]: 😮 Thank you for such an comprehensive documentation! @1orZero

**Review by thejackwu — APPROVED** [review]

**Pull request merged**